src/main/java/com/prgrms/catpicture/repository/CatPictureRepository.java

```diff
@@ -0,0 +1,22 @@
+package com.prgrms.catpicture.repository;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import com.prgrms.catpicture.model.CatPicture;
+
+public interface CatPictureRepository extends JpaRepository<CatPicture, Long> {
+	Optional<CatPicture> findByImageId(String imageId);
+
+	boolean existsByImageId(String imageId);
+
+	List<CatPicture> findAllByName(String name);
+
+	@Query(value = "select id, height, image_id, name, origin, temperament, url, width "
+		+ "from cat_picture as c order by RAND() limit 50",
+		nativeQuery = true)
```

Comment:
```suggestion
	@Query(value = "select * "
		+ "from cat_picture order by RAND() limit 50",
		nativeQuery = true)
```

모든 컬럼을 다 가져올거면 *를 사용해도 좋다고 생각합니다.

Reply:
감사합니다!!

Reply:
아 그리고 건우님은 랜덤 50개를 추출하는 로직을 어떻게 구현하셨나요? 저는 보신 것처럼 네이티브 쿼리를 써서 Rand()를 이용해서 50개를 가져왔는데 이 방법이 인덱스를 타지 않아서 성능적으로 안좋다고 들어서 혹시 다른 방법으로 구현하셨을까요??

Reply:
저도 고려하지 못했네요. 저도 수영님의 쿼리랑 같습니다.
밑에 글을 보고 같이 해결해봐요.
저도 쿼리 튜닝에 대해 공부해 봐야겠어요.
[ORDER BY RAND 쿼리 튜닝](https://blog.naver.com/sinjoker/221524576602)